lib/models/release.ts

```diff
@@ -389,6 +389,56 @@ const getReleaseModel = function (
 		);
 	}
 
+	/**
+	 * @summary Promotes a 'draft' release to 'final' (non-draft)
+	 * @name finalize
+	 * @public
+	 * @function
+	 * @memberof balena.models.release
+	 *
+	 * @param {String|Number} commitOrId - release commit (string) or id (number)
+	 * @fulfil {void}
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.release.finalize(123).then(function() {
+	 * 	console.log('finalized!');
+	 * });
+	 *
+	 * @example
+	 * balena.models.release.finalize('7cf02a6').then(function() {
+	 * 	console.log('finalized!');
+	 * });
+	 *
+	 */
+	async function finalize(commitOrId: string | number): Promise<void> {
+		if (commitOrId == null) {
+			throw new errors.BalenaReleaseNotFound(commitOrId);
+		}
+
+		if (isId(commitOrId)) {
+			await pine.patch<Release>({
+				resource: 'release',
+				id: commitOrId,
+				body: {
+					is_final: true,
+				},
+			});
+		} else {
+			await pine.patch<Release>({
+				resource: 'release',
+				body: {
+					is_final: true,
+				},
```

Comment:
NIT: In general we prefer having the `body` after the `options` so that we always first have to think what the `.patch` is going to affect and only after that think which props are going to change.